bus-mapping/src/evm/opcodes/codesize.rs

```diff
@@ -0,0 +1,107 @@
+use crate::{
+    circuit_input_builder::{CircuitInputStateRef, ExecStep},
+    Error,
+};
+
+use eth_types::GethExecStep;
+
+use super::Opcode;
+
+#[derive(Clone, Copy, Debug)]
+pub(crate) struct Codesize;
+
+impl Opcode for Codesize {
+    fn gen_associated_ops(
```

Comment:
If we need to implement error handling for `code >24kB` then we cannot reuse `StackOnlyOpcode` no?

Reply:
I have handled this in the latest commit :)

Reply:
I am a bit confused by this error handling.  As far as I understand, the limit on code size is enforced at contract creation (see https://github.com/ethereum/EIPs/blob/master/EIPS/eip-170.md#specification), so all deployed code must be valid, and thus a `CODESIZE` should never fail because of the actual code size.
So I think this error handling can't be triggered in a normal setup.

Did I miss something?

Reply:
@ed255 yes makes sense. After looking more carefully, yes it should be handled at the time of CREATE or CREATE2, and codesize opcode can simply use the stack only opcode method. I'll make the change.

I'll also remove the err and constant I added since it is not appropriate to add along with this PR.

Reply:
I have reverted the changes I made for max code size handling. What do you think we can do to keep the assertion? In that case we can't use the `StackOnlyOpcode` for `CODESIZE`

Reply:
> Although, I would like to have the assertion that the code size from state matches the one from the next step's stack output. In this case, I won't be able to use the `StackOnlyOpcode` right?

I like having assertions like this one!  They can be extremely useful to figure out bugs that we may encounter in the future.  I don't think using the `StackOnlyOpcode` is compatible with having that assertion; and for that I think it's OK to leave the `Codesize::gen_associated_ops` implementation.  So from my side this resolves the conversation :)

Reply:
My bad.. you're right!!!!